jablib/src/test/java/org/jabref/logic/importer/fileformat/ReferImporterTest.java

```diff
@@ -0,0 +1,158 @@
+package org.jabref.logic.importer.fileformat;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.StringReader;
+import java.net.URISyntaxException;
+import java.nio.file.Path;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import org.jabref.logic.util.StandardFileType;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.field.StandardField;
+import org.jabref.model.entry.types.StandardEntryType;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class ReferImporterTest {
+
+    private ReferImporter referImporter;
+
+    @BeforeEach
+    void setUp() {
+        referImporter = new ReferImporter();
+    }
+
+    @Test
+    void getFormatName() {
+        assertEquals("Refer/BibIX", referImporter.getName());
+    }
+
+    @Test
+    void getCLIId() {
+        assertEquals("refer-bibIX", referImporter.getId());
+    }
+
+    @Test
+    void getDescription() {
+        assertEquals("Import for the Refer/BibIX file.", referImporter.getDescription());
+    }
+
+    @Test
+    void getExtension() {
+        assertEquals(StandardFileType.TXT, referImporter.getFileType());
+    }
+
+    @Test
+    void isRecognizedFormat() throws IOException, URISyntaxException {
+        List<String> list = Arrays.asList("refer.bibIX.Journal.ref", "refer.bibIX.ref");
+
+        for (String str : list) {
+            Path file = Path.of(ReferImporterTest.class.getResource(str).toURI());
+            assertTrue(referImporter.isRecognizedFormat(file));
+        }
+    }
+
+    @Test
+    void isRecognizedFormatReject() throws IOException, URISyntaxException {
+        List<String> list = Arrays.asList("Endnote.pattern.A.enw", "IEEEImport1.txt", "IsiImporterTest1.isi", "IsiImporterTestInspec.isi",
+                "IsiImporterTestWOS.isi", "IsiImporterTestMedline.isi", "RisImporterTest1.ris",
+                "Endnote.pattern.no_enw", "empty.pdf", "pdf/annotated.pdf");
+
+        for (String str : list) {
+            Path file = Path.of(ReferImporterTest.class.getResource(str).toURI());
+            assertFalse(referImporter.isRecognizedFormat(file));
+        }
+    }
+
+    @Test
+    void importDatabaseSingleEntryTest() throws IOException, URISyntaxException {
+        Path file = Path.of(ReferImporterTest.class.getResource("refer.bibIX.Journal.ref").toURI());
+        List<BibEntry> bibEntryList = referImporter.importDatabase(file).getDatabase().getEntries();
+        BibEntry entry = bibEntryList.getFirst();
+
+        assertEquals(1, bibEntryList.size());
+        assertEquals(StandardEntryType.Article, entry.getType());
```

Comment:
Easier to create the expexected bibEntry and compare the whole entry. See how it's done in other tests